tests/test_iotedgedev.py

```diff
@@ -1,266 +1,385 @@
+        # https://github.com/Azure/iotedgedev/issues/346
+        if (template == "nodejs") and (platform.system().lower() != 'windows'):
+            continue
+        else:
+            result = create_solution(template)
+            assert_solution_folder_structure(template)
+            assert 'AZURE IOT EDGE SOLUTION CREATED' in result.output
+            clean_folder(test_solution_dir)
+
+
+def test_solution_build_with_default_platform(prepare_solution_with_env):
+    result = runner_invoke(['build'])
+
+    module_name = "filtermodule"
+    test_solution_config_dir = os.path.join('config', 'deployment.' + get_platform_type() + '.json')
+    env_container_registry_server = os.getenv("CONTAINER_REGISTRY_SERVER")
+
+    assert 'BUILD COMPLETE' in result.output
+    assert 'ERROR' not in result.output
+    assert env_container_registry_server + "/" + module_name + ":0.0.1-" + get_platform_type() in json.load(open(test_solution_config_dir))[
+        "modulesContent"]["$edgeAgent"]["properties.desired"]["modules"][module_name]["settings"]["image"]
+    assert module_name in get_all_docker_images()
+
+
+@pytest.mark.skipif(get_docker_os_type() == 'windows', reason='Debugger does not support C# in windows container')
+def test_solution_build_with_debug_template():
+    os.chdir(test_solution_shared_lib_dir)
+
+    result = runner_invoke(['build', '-f', os.environ["DEPLOYMENT_CONFIG_DEBUG_TEMPLATE_FILE"], '-P', get_platform_type()])
+
+    module_name = "sample_module"
+    test_solution_shared_debug_config = os.path.join('config', 'deployment.debug.' + get_platform_type() + '.json')
+    env_container_registry_server = os.getenv("CONTAINER_REGISTRY_SERVER")
+
+    assert 'BUILD COMPLETE' in result.output
+    assert 'ERROR' not in result.output
+    assert env_container_registry_server + "/" + module_name + ":0.0.1-" + get_platform_type() + ".debug" in json.load(open(test_solution_shared_debug_config))[
+        "modulesContent"]["$edgeAgent"]["properties.desired"]["modules"][module_name]["settings"]["image"]
+    assert module_name in get_all_docker_images()
+
+
+def test_solution_push_with_default_platform(prepare_solution_with_env):
+    result = runner_invoke(['push'])
+
+    module_name = "filtermodule"
+    test_solution_config_dir = os.path.join('config', 'deployment.' + get_platform_type() + '.json')
+    env_container_registry_server = os.getenv("CONTAINER_REGISTRY_SERVER")
+
+    assert 'BUILD COMPLETE' in result.output
+    assert 'PUSH COMPLETE' in result.output
+    assert 'ERROR' not in result.output
+    assert env_container_registry_server + "/" + module_name + ":0.0.1-" + get_platform_type() in json.load(open(test_solution_config_dir))[
+        "modulesContent"]["$edgeAgent"]["properties.desired"]["modules"][module_name]["settings"]["image"]
+    assert module_name in get_all_docker_images()
+
+
+def test_generate_deployment_manifest():
+    test_solution_build_with_platform()
```

Comment:
Question: why do we need to call `test_solution_build_with_platform` here?

Reply:
I need a original deployment file to compare with a newer deployment file.

Reply:
The build command basically includes the same code logic of genconfig, so the comparison here does not make too much sense.

Reply:
@neil-yechenwei Is this resolved?

Reply:
Updated code